### PR TITLE
Migrate Board roles to user_roles

### PR DESCRIPTION
**Pull request by danieljames-dj**

### CODE REVIEW

**Comment by gregorbg** [comment]
Code looks okay but you're getting actual test failures.
Please fix them before I submit a formal approval.

File: app/models/user.rb [hunk 1]
```diff
@@ -1297,7 +1297,8 @@ def is_special_account?
       any_kind_of_delegate? ||
       !delegated_competitions.empty? ||
       !competitions_announced.empty? ||
-      !competitions_results_posted.empty?
+      !competitions_results_posted.empty? ||
+      board_member?
```
Comment (gregorbg) [inline app/models/user.rb:1301]: Why is this change necessary? Feels like you're touching a method that should not be affected by your migration

File: app/webpacker/components/Panel/Board/BoardEditorPage/BoardEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          ))}
+        </Table.Body>
+      </Table>
+      {openModal && (
+      <Modal
+        size="fullscreen"
+        open={openModal}
+        onClose={() => {
+          setOpenModal(false);
+          newBoardRole(null);
```
Comment (gregorbg) [inline app/webpacker/components/Panel/Board/BoardEditorPage/BoardEditor.jsx:56]: Shouldn't this be `setNewBoardRole`? In that case, you can extract it into its own function and use that new function here and also in the `onSubmit` success fn below.

File: db/migrate/20240326023855_create_board_roles.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      user_id: team_member.user_id,
+      group_id: group.id,
+      start_date: team_member.start_date,
+      end_date: team_member.end_date,
+    )
+  end
+
+  def change
+    board_group = UserGroup.board_group
+    Team.c_find_by_friendly_id!('board').team_members.each do |member|
```
Comment (gregorbg) [inline db/migrate/20240326023855_create_board_roles.rb:15]: Shouldn't `Team.board` do the trick?

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (danieljames-dj) [reply app/models/user.rb:1301]: Actually board member is also considered as special account according to unit test. Earlier, this was covered in `self.teams.any`, but now since board is out of teams, this is necessary..

File: app/webpacker/components/Panel/Board/BoardEditorPage/BoardEditor.jsx (hunk 2 above)
Comment (danieljames-dj) [reply app/webpacker/components/Panel/Board/BoardEditorPage/BoardEditor.jsx]: Oh yes, done. Thanks.

File: db/migrate/20240326023855_create_board_roles.rb (hunk 3 above)
Comment (danieljames-dj) [reply db/migrate/20240326023855_create_board_roles.rb:15]: Actually I was thinking to remove `self.board` from `team.rb` as it is no longer used, but I missed to remove it. I've removed it from `team.rb` now.

File: app/models/user.rb (hunk 1 above)
Comment (gregorbg) [reply app/models/user.rb:1301]: Aha! Makes sense, thanks for the explanation.

**Review by gregorbg — APPROVED** [review]
